app/handlers/handlers_resource.py

```diff
@@ -21,16 +21,20 @@ def twingate_resource_create(body, spec, memo, logger, patch, **kwargs):
     )
 
 
-@kopf.on.update("twingateresource")
-def twingate_resource_update(old, new, diff, status, memo, logger, **kwargs):
+@kopf.on.update("twingateresource", field="spec")
+def twingate_resource_update(spec, new, diff, status, memo, logger, **kwargs):
     logger.info(
         "Got TwingateResource update request: %s. Diff: %s. Status: %s.",
         new,
         diff,
         status,
     )
 
-    crd = ResourceSpec(**new["spec"])
+    crd = ResourceSpec(**spec)
+
+    if len(diff) == 1 and diff[0][0] == "add" and diff[0][1] == ("id",):
```

Comment:
nit: I wonder if there is a nicer way to do this because the indexes are quite abstract. Maybe
```python
if diff == (("add", ("id",), None, ANY),):
```
? 🤔

Reply:
@minhtule I thought about it but `ANY` is in `unittest` which I wouldnt want to import on non testing code